utils/zerovec/src/varzerovec/components.rs

```diff
@@ -578,23 +613,47 @@ where
         dat_offset = dat_limit;
     }
 
-    debug_assert_eq!(
-        idx_offset,
-        LENGTH_WIDTH + METADATA_WIDTH + F::INDEX_WIDTH * elements.len()
-    );
+    debug_assert_eq!(idx_offset, METADATA_WIDTH + F::INDEX_WIDTH * elements.len());
     assert_eq!(dat_offset, output.len());
 }
 
-pub fn compute_serializable_len<T, A, F>(elements: &[A]) -> Option<u32>
+/// Writes the bytes for a VarZeroSlice into an output buffer.
+///
+/// Every byte in the buffer will be initialized after calling this function.
+///
+/// # Panics
+///
+/// Panics if the buffer is not exactly the correct length.
+pub fn write_serializable_bytes<T, A, F>(elements: &[A], output: &mut [u8])
```

Comment:
Thought: It might be nice to refactor this code into an `EncodeAsVarULE` impl, which has well-defined safety requirements and reduces indirection (not now, maybe later)

Reply:
Not sure that makes sense, I think this is the thing that underlies the EncodeAsVarULE infrastructure. But maybe?

Reply:
This is VarZeroVecComponents; which is extremely low level. components.rs is basically where all of the actual VZV format algorithms live.